### PR TITLE
Api integration updated for Intune scep

### LINKED ISSUE #6933: PFPKI: Cloud integration with Intune not working
**Describe the bug**
Using the cloud integration option in  SCEP settings in a PKI template seems to break SCEP functionality for that template.

**To Reproduce**
Configure the PKI as per the steps in the manual. Configure an Intune cloud connector. Do not enable this connector in the SCEP settings in the template:

image

SCEP functionality works, both locally and on a Windows 10 laptop:

image

Turn on the cloud integration in the template:

image

SCEP functionality is broken, both locally and on a Windows 10 laptop:

image

**Expected behavior**
Cloud functionality should work on a template, to be able to provision Intune managed devices with certificates to be used with EAP-TLS.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 99

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Update API call and authentication for intune. (based on the work on https://github.com/inverse-inc/packetfence/pull/7115)

Azure AD API has been deprecated in favour of MS Graph API.
Adal authentication will be deprecated in favour of MSAL.

# Impacts
Intune scep integration

# Issue
fixes #6933

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Bug Fixes
* Updates Intune integration to new Microsoft Graph API and MSAL Authentication (https://github.com/inverse-inc/packetfence/issues/6933)

# UPGRADE file entries

New permissions needs to be updated on the Azure side for the PacketFence application ([See documentation](https://www.packetfence.org/doc/PacketFence_Installation_Guide.html#_azure_configuration))

### CODE REVIEW

**Review by julsemaan — COMMENTED** [review]
I'm good with the code.

The upgrade notes just need to be a bit clearer and have more 🥩 but I can polish this during the merge

File: go/caddy/pfpki/cloud/intune.go [hunk 1]
```diff
@@ -107,35 +109,21 @@ func (cl *Intune) NewCloud(ctx context.Context, name string) error {
...
-
-	// Intune token
-
-	spt, err := adal.NewServicePrincipalToken(*oauthConfig, cl.ClientID, cl.ClientSecret, graphResourceUrl)
-
-	err = spt.Refresh()
-
-	var token adal.Token
-
+	cred, err := azidentity.NewClientSecretCredential(cl.TenantID, cl.ClientID, cl.ClientSecret, nil)
```
Comment (jrouzierinverse) [inline go/caddy/pfpki/cloud/intune.go:112]: Is it safe to ignore this error?

File: go/caddy/pfpki/cloud/intune.go [hunk 2]
```diff
@@ -107,35 +109,21 @@ func (cl *Intune) NewCloud(ctx context.Context, name string) error {
...
+	// Fetch the token for Graph api
+	tk, err := cred.GetToken(
+		context.TODO(), policy.TokenRequestOptions{Scopes: []string{msGraphResourceUrl + ".default"}},
+	)
 	if err == nil {
-		token = spt.Token()
-		cl.AccessToken = "Bearer " + token.AccessToken
+		cl.AccessToken = "Bearer " + tk.Token
+	} else {
+		log.Print(err)
```
Comment (jrouzierinverse) [inline go/caddy/pfpki/cloud/intune.go:124]: Should we return this error?

File: go/caddy/pfpki/cloud/intune.go [hunk 3]
```diff
@@ -195,22 +183,27 @@ func (cl *Intune) NewCloud(ctx context.Context, name string) error {
...
+
+	// Fetch the token for intune api
+	tk, err = cred.GetToken(
+		context.TODO(), policy.TokenRequestOptions{Scopes: []string{intuneResourceUrl + "/.default"}},
+	)
+
+	if err == nil {
+		cl.AccessToken = "Bearer " + tk.Token
+	} else {
+		log.Print(err)
```
Comment (jrouzierinverse) [inline go/caddy/pfpki/cloud/intune.go:209]: Should we return this error?

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by trs80** [comment]
MS extended Azure AD Graph support until June 30, 2023 https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/microsoft-entra-change-announcements-september-2022-train/ba-p/2967454

File: go/caddy/pfpki/cloud/intune.go (hunk 2 above)
Comment (fdurand) [reply go/caddy/pfpki/cloud/intune.go:124]: Yes, for troubleshooting purpose it´s really usefull.

File: go/caddy/pfpki/cloud/intune.go (hunk 3 above)
Comment (fdurand) [reply go/caddy/pfpki/cloud/intune.go:209]: Yes, for troubleshooting purpose it´s really usefull.

**Comment by fdurand** [comment]
> MS extended Azure AD Graph support until June 30, 2023 https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/microsoft-entra-change-announcements-september-2022-train/ba-p/2967454

in the azure portal it´s more complex now to add the old permissions, so it´s better to force the PacketFence setup to use the new way.

**Comment by trs80** [comment]
> > MS extended Azure AD Graph support until June 30, 2023 https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/microsoft-entra-change-announcements-september-2022-train/ba-p/2967454
> 
> in the azure portal it´s more complex now to add the old permissions, so it´s better to force the PacketFence setup to use the new way.

Oh yeah, just saying the notice [here](https://github.com/inverse-inc/packetfence/pull/7222/files#diff-4bfc870087f335c94e37a38db6e6b35fc9a7637ca58cc8150be43c661faa005bR218) could be updated. Perhaps something more like
> Previous versions of PacketFence used Azure Active Directory Graph API which is now deprecated and will stop working after June 2023. Since 12.x you will need to add the new Microsoft Graph API permissions instead as PacketFence no longer uses the old API.

**Comment by julsemaan** [comment]
@jrouzierinverse, can you do another round of review with @fdurand's latest comments

**Comment by jrouzierinverse** [comment]
rebase

**Comment by fdurand** [comment]
Fixed and rebased

**Comment by fdurand** [comment]
BUMP

**Pull request merged**